cloudsync/smartsync.py

```diff
@@ -427,3 +427,15 @@ def smart_delete_path(self, local_oid, local_path):
                 self.state.update_entry(ent, LOCAL, local_oid, path=local_path, changed=True, exists=False)
                 self.state.requestset.add(ent)
                 self.state.excludeset.discard(ent)
+
+    def smart_rename(self, side, oid, new_path) -> str:
```

Comment:
I'm wondering if we need an async endpoint as well for directory renames. Probably not necessary right this instant and I'm not even sure if we support directory renames for smart sync right now

Reply:
Do we have any async endpoints? That's a bigger discussion. Perhaps the async code is the responsibility of the client

Reply:
Do we have any async endpoints? I think the question is whether async code belongs in the library or in the client code